assets/src/editor/events/tickets/editor-ticket/price-calculator/ticket-price-calculator-form-modal.js

```diff
@@ -0,0 +1,210 @@
+	// 				select( 'eventespresso/core' )
+	// 					.getRelatedEntities( ticket, 'prices' ) :
+	// 				[];
+	// 			console.log(
+	// 				'TicketPriceCalculatorFormModal.getTicketPrices() prices',
+	// 				prices
+	// 			);
+	// 			if ( prices ) {
+	// 				resolve( {
+	// 					ticket: ticket,
+	// 					prices: prices,
+	// 				} );
+	// 			}
+	// 		}
+	// 	);
+	// };
+
+	/**
+	 * @function
+	 * @return {Object} data
+	 */
+	loadHandler = async () => {
+		if (
+			isModelEntityOfModel( this.props.ticket, 'ticket' ) &&
+			Array.isArray( this.props.prices )
+		) {
+			return {
+				ticket: this.props.ticket,
+				prices: this.props.prices,
+			};
+		}
+		// return this.getTicket( this.props.ticketId )
+		// 	.then( this.getTicketPrices )
+		// 	.then( ( data ) => {
+		// 		return data;
+		// 	} );
+		// const ticket = await this.getTicket( this.props.ticketId );
+		// if ( ticket ) {
+		// 	console.log(
+		// 		'TicketPriceCalculatorFormModal.loadHandler() ticket',
+		// 		ticket
+		// 	);
+		// 	const prices = await this.getTicketPrices( ticket );
+		// 	if ( prices ) {
+		// 		console.log(
+		// 			'TicketPriceCalculatorFormModal.loadHandler() prices',
+		// 			prices
+		// 		);
+		// 		return {
+		// 			ticket: ticket,
+		// 			prices: prices,
+		// 		};
+		// 	}
+		// }
+	};
+
+	/**
+	 * @function
+	 * @param {Object} data
+	 */
+	submitHandler = async ( data ) => {
+		console.log( 'TicketPriceCalculatorFormModal.submitHandler()' );
+		console.log( ' >>> SUBMITTING DATA <<<', data );
+		// const ticket = ticketEntityFormSubmitHandler(
+		// 	this.state.ticket,
+		// 	data
+		// );
+		// console.log( ' >>> UPDATED TICKET <<<', ticket );
+		// this.setState( { ticket: ticket } );
+		this.toggleEditor();
+	};
+
+	/**
+	 * @function
+	 * @param {Object} event
+	 */
+	resetHandler = ( event ) => {
+		console.log( 'TicketPriceCalculatorFormModal.resetHandler()' );
+		console.log( ' >>> FORM RESET <<<', event );
+		// this.setState( { ticket: this.state.originalTicket } );
+	};
+
+	render() {
+		console.log( 'TicketPriceCalculatorFormModal.render()', this.props );
+		return (
+			<TicketPriceCalculatorForm
+				loadHandler={ this.loadHandler }
+				submitHandler={ this.submitHandler }
+				resetHandler={ this.resetHandler }
+				{ ...this.props }
+			/>
+		);
+	}
+}
+
+/**
+ * Enhanced TicketPriceCalculatorForm with Modal
+ */
+export default withEditorModal( {
+	title: __( 'Ticket Price Calculator', 'event_espresso' ),
+	customClass: 'ee-ticket-price-calculator-modal',
+	closeButtonLabel: __( 'close ticket price calculator', 'event_espresso' ),
+} )(
+	withSelect( ( select, ownProps ) => {
+		const ticket = select( 'eventespresso/core' )
+			.getTicketById( ownProps.ticketId );
+		const prices = isModelEntityOfModel( ticket, 'ticket' ) ?
+			select( 'eventespresso/core' )
+				.getRelatedEntities( ticket, 'prices' ) :
+			[];
+		return {
+			ticket,
+			prices,
+		};
+	} )( TicketPriceCalculatorFormModal )
```

Comment:
To get the number of renders down, I changed the withSelect to this:

```js
/**
 * Enhanced TicketPriceCalculatorForm with Modal
 */
export default withEditorModal( {
	title: __( 'Ticket Price Calculator', 'event_espresso' ),
	customClass: 'ee-ticket-price-calculator-modal',
	closeButtonLabel: __( 'close ticket price calculator', 'event_espresso' ),
} )(
	withSelect( ( select, ownProps ) => {
		const ticket = select( 'eventespresso/core' )
			.getTicketById( ownProps.ticketId );
		const prices = isModelEntityOfModel( ticket, 'ticket' ) ?
			select( 'eventespresso/core' )
				.getRelatedEntities( ticket, 'prices' ) :
			[];
		return {
			ticket,
			prices: isEmpty( prices ) ? DEFAULT_EMPTY_ARRAY : prices,
		};
	} )( TicketPriceCalculatorFormModal )
);
```

You'd need to `import { isEmpty } from 'lodash'` and define a `DEFAULT_EMPTY_ARRAY` somewhere in the upper scope.

I don't think similar treatment needs done to `ticket` because it should be always returning either `null` or the same ticket `BaseEntity` instance.